tests/manage/nfs_feature/test_nfs_feature_enable_for_ODF_clusters.py

```diff
@@ -1129,174 +1129,181 @@ def test_respin_app_pod_exported_nfs_volume_incluster(
+                pod_obj.name, 60
+            ), f"Pod {pod_obj.name} is not deleted"
 
-        respinned_pod_objs = pod.get_all_pods(
-            namespace=self.namespace, selector=["nfs-test-pod"], selector_label="name"
-        )
+            assert self.pod_obj.wait_for_resource(
+                resource_count=1,
+                condition=constants.STATUS_RUNNING,
+                selector="name=nfs-test-pod",
+                dont_allow_other_resources=True,
+                timeout=60,
+            )
 
-        respinned_pod_obj = respinned_pod_objs[0]
-        log.info(f"pod obj name----{respinned_pod_obj.name}")
+            respinned_pod_objs = pod.get_all_pods(
+                namespace=self.namespace,
+                selector=["nfs-test-pod"],
+                selector_label="name",
+            )
 
-        # Able to read the /mnt/test file's content from inside the respined pod
-        command = "bash -c " + '"cat ' + ' /mnt/test"'
-        result = respinned_pod_obj.exec_cmd_on_pod(
-            command=command,
-            out_yaml_format=False,
-        )
-        assert result.rstrip() == "Before respin"
+            respinned_pod_obj = respinned_pod_objs[0]
+            log.info(f"pod obj name----{respinned_pod_obj.name}")
 
-        # Edit /mnt/test file
-        command = "bash -c " + '"echo ' + "'After respin'" + '  >> /mnt/test"'
+            # Able to read the /mnt/test file's content from inside the respined pod
+            command = "bash -c " + '"cat ' + ' /mnt/test"'
+            result = respinned_pod_obj.exec_cmd_on_pod(
+                command=command,
+                out_yaml_format=False,
+            )
+            assert result.rstrip() == "Before respin"
 
-        respinned_pod_obj.exec_cmd_on_pod(
-            command=command,
-            out_yaml_format=False,
-        )
-        # Able to read updated /mnt/test file
-        command = "bash -c " + '"cat ' + ' /mnt/test"'
-        result = respinned_pod_obj.exec_cmd_on_pod(
-            command=command,
-            out_yaml_format=False,
-        )
-        assert result.rstrip() == "Before respin" + """\n""" + "After respin"
+            # Edit /mnt/test file
+            command = "bash -c " + '"echo ' + "'After respin'" + '  >> /mnt/test"'
 
-        # Delete deployment config
-        cmd_delete_deployment_config = "delete dc nfs-test-pod"
-        self.storage_cluster_obj.exec_oc_cmd(cmd_delete_deployment_config)
+            respinned_pod_obj.exec_cmd_on_pod(
+                command=command,
+                out_yaml_format=False,
+            )
+            # Able to read updated /mnt/test file
+            command = "bash -c " + '"cat ' + ' /mnt/test"'
+            result = respinned_pod_obj.exec_cmd_on_pod(
+                command=command,
+                out_yaml_format=False,
+            )
+            assert result.rstrip() == "Before respin" + """\n""" + "After respin"
+        except Exception as ex:
```

Comment:
This is a very broad exception, I believe this can be skipped and the test can simply have try and finally block?